base/data/src/main/java/org/openscience/cdk/DefaultChemObjectBuilder.java

```diff
@@ -117,99 +117,136 @@ private static boolean getSystemProp(final String key, final boolean defaultValu
+    private enum Holder implements IChemObjectBuilder {
 
-    private DefaultChemObjectBuilder() {
+        INSTANCE;
+
+        private final DynamicFactory factory = new DynamicFactory(200);
 
         // self reference required for stereo-elements
         final IChemObjectBuilder self = this;
 
-        // elements
-        factory.register(IAtom.class, Atom.class);
-        factory.register(IPseudoAtom.class, PseudoAtom.class);
-        factory.register(IElement.class, Element.class);
-        factory.register(IAtomType.class, AtomType.class);
-        factory.register(IFragmentAtom.class, FragmentAtom.class);
-        factory.register(IPDBAtom.class, PDBAtom.class);
-        factory.register(IIsotope.class, Isotope.class);
+        Holder() {
+            // elements
+            factory.register(IAtom.class, Atom.class);
+            factory.register(IPseudoAtom.class, PseudoAtom.class);
+            factory.register(IElement.class, Element.class);
+            factory.register(IAtomType.class, AtomType.class);
+            factory.register(IFragmentAtom.class, FragmentAtom.class);
+            factory.register(IPDBAtom.class, PDBAtom.class);
+            factory.register(IIsotope.class, Isotope.class);
 
-        // electron containers
-        factory.register(IBond.class, Bond.class);
-        factory.register(IElectronContainer.class, ElectronContainer.class);
-        factory.register(ISingleElectron.class, SingleElectron.class);
-        factory.register(ILonePair.class, LonePair.class);
+            // electron containers
+            factory.register(IBond.class, Bond.class);
+            factory.register(IElectronContainer.class, ElectronContainer.class);
+            factory.register(ISingleElectron.class, SingleElectron.class);
+            factory.register(ILonePair.class, LonePair.class);
 
-        // atom containers
-        if (CDK_LEGACY_AC) {
-            System.err.println("[WARN] Using the old AtomContainer implementation.");
-            factory.register(IAtomContainer.class, AtomContainer.class);
-        } else {
-            factory.register(IAtomContainer.class, AtomContainer2.class);
-        }
-        factory.register(IRing.class, Ring.class);
-        factory.register(ICrystal.class, Crystal.class);
-        factory.register(IPolymer.class, Polymer.class);
-        factory.register(IPDBPolymer.class, PDBPolymer.class);
-        factory.register(IMonomer.class, Monomer.class);
-        factory.register(IPDBMonomer.class, PDBMonomer.class);
-        factory.register(IBioPolymer.class, BioPolymer.class);
-        factory.register(IPDBStructure.class, PDBStructure.class);
-        factory.register(IAminoAcid.class, AminoAcid.class);
-        factory.register(IStrand.class, Strand.class);
+            // atom containers
+            if (CDK_LEGACY_AC) {
+                System.err.println("[WARN] Using the old AtomContainer implementation.");
```

Comment:
Should this not use the logger?

Reply:
I think is okay as about to remove the Legacy Atom Container anyways

Reply:
(was already like this)

Reply:
oh, sorry I missed that.